jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassStatistics.java

```diff
@@ -45,69 +45,85 @@ public static long maxSizeBytes() {
-    }
-    
-    /**
-     * Returns the maximum space allowed for AOT data of the cache that the JVM is currently
-     * connected to.
-     * 
-     * @return the maximum shared classes cache space allowed for AOT data in bytes or -1 if it is not set.
-     */
-    public static long maxAotBytes() {
-    	return maxAotBytesImpl();
-    }
-    
-    /**
-     * Returns the minimum space reserved for JIT data of the cache that the JVM is currently
-     * connected to.
-     * 
-     * @return the minimum shared classes cache space reserved for JIT data in bytes or -1 if it is not set.
-     */
-    public static long minJitDataBytes() {
-    	return minJitDataBytesImpl();
-    }
-    
-    /**
-     * Returns the maximum space allowed for JIT data of the cache that the JVM is currently
-     * connected to.
-     * 
-     * @return the maximum shared classes cache space allowed for JIT data in bytes  or -1 if it is not set.
-     */
-    public static long maxJitDataBytes() {
-    	return maxJitDataBytesImpl();
-    }
+	 * Returns the minimum space reserved for AOT data of the cache that the JVM is currently
+	 * connected to.
+	 *
+	 * @return the minimum shared classes cache space reserved for AOT data in bytes or -1 if it is not set.
+	 */
+	public static long minAotBytes() {
+		return minAotBytesImpl();
+	}
 
-	
-	private static native long maxSizeBytesImpl();
+	/**
+	 * Returns the maximum space allowed for AOT data of the cache that the JVM is currently
+	 * connected to.
+	 *
+	 * @return the maximum shared classes cache space allowed for AOT data in bytes or -1 if it is not set.
+	 */
+	public static long maxAotBytes() {
+		return maxAotBytesImpl();
+	}
+
+	/**
+	 * Returns the minimum space reserved for JIT data of the cache that the JVM is currently
+	 * connected to.
+	 *
+	 * @return the minimum shared classes cache space reserved for JIT data in bytes or -1 if it is not set.
+	 */
+	public static long minJitDataBytes() {
+		return minJitDataBytesImpl();
+	}
+
+	/**
+	 * Returns the maximum space allowed for JIT data of the cache that the JVM is currently
+	 * connected to.
+	 *
+	 * @return the maximum shared classes cache space allowed for JIT data in bytes  or -1 if it is not set.
+	 */
+	public static long maxJitDataBytes() {
+		return maxJitDataBytesImpl();
+	}
+
+	/**
+	 * Returns the SysV shmem nattch value for a non-persistent top level cache.
+	 * On Windows or for persistent caches, -1 is returned. Depending on the platform
+	 * and OS, the value may or may not indicate the number of JVMs attached to the cache.
+	 *
+	 * @return the SysV shmem nattch value for a top level non-persistent cache, or -1 if it cannot be determined.
+	 */
+	public static long numberAttached() {
+		return nattachImpl();
+	}
 
+	/**
+	 * Returns the full cache path and filename.
+	 *
+	 * @return the full cache path and filename, or NULL if it cannot be determined.
+	 */
+	public static String pathAndFilename() {
+		return cacheDirImpl();
```

Comment:
I think this should just be `path()` (or `getPath()`); it answers the _path_ of the cache file.

Reply:
How about cachePath()? path() would be ok but more cryptic. getPath() doesn't fit with the other methods as none of them use `get`.

Reply:
`cachePath()` works for me. I agree that using the `get` prefix would be inconsistent with other, existing accessor methods.

Reply:
The javadoc should be updated: it doesn't return 'path and filename', just a pathname.